Modules/Promotion/Http/Services/PromotionService.php

```diff
@@ -182,7 +232,10 @@ public function delete(string $promotionKey): array
             ];
         }
 
-        $promotion->delete();
+        $promotion->status = Promotion::STATUS_DEACTIVATED;
```

Comment:
@arijit1111 at this line: `$promotion = Promotion::where('promotion_key', $promotionKey)->first();` we should be validating if the promotion really belongs to the active user.

Reply:
applied in controller, please have a look

Reply:
@arijit1111 I still cannot see where you have applied.

Reply:
@arijit1111 Thanks, I can see as its been validated through controller